src/crypto/equihash.cpp

```diff
@@ -22,19 +22,24 @@
 template<unsigned int N, unsigned int K>
 int Equihash<N,K>::InitialiseState(eh_HashState& base_state)
 {
-    unsigned int n = N;
-    unsigned int k = K;
     unsigned char personalization[crypto_generichash_blake2b_PERSONALBYTES] = {};
-    memcpy(personalization, "ZcashPOW", 8);
-    memcpy(personalization+8,  &n, 4);
-    memcpy(personalization+12, &k, 4);
+    memcpy(personalization, "ZcashPoW", 8);
+    personalization[8]  =  N        & 0xFF;
```

Comment:
Or better:

```
uint32_t n = htole32(N);
uint32_t k = htole32(K);
mcpy(personalization+8,  &n, 4);
memcpy(personalization+12,  &k, 4);
```

This matches [how the indices get serialised in the block header](https://github.com/zcash/zcash/blob/447444ae7a3ed4e13bfb09e0c15d458070e9b904/src/serialize.h#L92).